superdesk/text_checkers/ai/semaphore.py

```diff
@@ -0,0 +1,472 @@
+                        continue  # Skip if there's a mismatch in the number of labels and GUIDs
+
+                    parent_qcode = None  # Track the parent qcode
+                    for label, guid in zip(labels, guids):
+                        tag_data = {
+                            "name": label,
+                            "qcode": guid,
+                            "parent": parent_qcode,
+                            "source": "Semaphore",
+                            "altids": {"source_name": "source_id"},
+                            "original_source": "original_source_value",
+                            "scheme": "http://cv.iptc.org/newscodes/mediatopic/"
+                        }
+                        add_to_dict("subject", tag_data)
+                        parent_qcode = guid  # Update the parent qcode for the next iteration
+
+                return response_dict
+                                          
+                
+            
+            json_response = transform_xml_response(root)
+
+            
+
+            return json_response
+
+        except requests.exceptions.RequestException as e:
+            traceback.print_exc()
+            logger.error(f"Semaphore request failed. We are in analyze RequestError exception: {str(e)}")
+
+        except Exception as e:
+            traceback.print_exc()
+            logger.error(f"An error occurred. We are in analyze exception: {str(e)}")
+
+    
+    def html_to_xml(self,html_content: str) -> str: 
+
+        def clean_html_content(input_str):
+            # Remove full HTML tags using regular expressions
+            your_string = input_str.replace('<p>', '')
+            your_string = your_string.replace('</p>', '')
+            your_string = your_string.replace('<br>', '')
+            your_string = your_string.replace('&nbsp;', '')
+            your_string = your_string.replace('&amp;', '')
+            your_string = your_string.replace('&lt;&gt;', '')
+            
+            
+
+            
+            return your_string    
+	   
+			
+        xml_template = """<?xml version="1.0" ?>
+				<request op="CLASSIFY">
+				<document>
+					<body>&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
+				&lt;story&gt;
+					&lt;headline&gt;{}&lt;/headline&gt;
+					&lt;headline_extended&gt;{}&lt;/headline_extended&gt;
+					&lt;body_html&gt;{}&lt;/body_html&gt; 
+                    &lt;slugline&gt;{}&lt;/slugline&gt;                             					
+				&lt;/story&gt;
+				</body>
+				</document>
+				</request>
+				"""
+
+				
+			
+			
+        body_html = html_content['body_html']
+        headline = html_content['headline']
+        headline_extended = html_content['abstract']
+        slugline = html_content['slugline']
+        
+	# Embed the 'body_html' into the XML template		
+        xml_output = xml_template.format(headline,headline_extended,body_html,slugline)
+        xml_output = clean_html_content(xml_output)
```

Comment:
you can use [get_text](https://github.com/superdesk/superdesk-core/blob/develop/superdesk/text_utils.py#L59) helper to convert html to text